active/0004-extensibility-support-for-deployment-stacks.md

```diff
@@ -0,0 +1,405 @@
+
+Here are factors that need to be considered for the rollout:
+- AKS/Kubernetes in Azure is only available in production environments and this is the only available V2 extension at
+this time that can be tested with.
+- New prod API versions require going through swagger review which can take a lot of time.
+- SDK updates for the deployment service should be consumed by Stacks before releasing the Stacks Stable API Version.
+
+The rollout:
+1. Changes in both the Deployments service and Stacks service will be feature gated.
+    - Feature gate needs to be at the subscription level to enable Canary-scoped development and testing.
+    - The feature gate should apply on whatever the current stable (or preview) API version is for each service 
+      (or all API versions if it easier).
+1. Development iteration of the Deployment service and Stack service will take place and be tested in Canary until the 
+milestone is hit.
+1. Develop some synthetics to run in production that use the kubernetes extension in a Deployment stack.
+1. Once the milestone implementation is sufficiently bug bashed and signed off by the team, the feature gate will be
+disabled.
+1. The Swagger update process will begin for at least a new Prod Preview API version.
+    - Priority should be given to the deployments side due to it being a dependency.
+    - Both swagger updates could be potentially be done in parallel.
+    - It may be possible to start the Deployment service swagger early if there's confidence in its changes.
+1. Incorporate any changes from swagger review and retest if needed using the feature flags.
+1. Merge the Deployment service swagger changes and release SDK.
+1. Update stacks code to consume new deployment SDK code.
+1. Merge the Deployment stack swagger changes and release SDK.
+1. Update client tools such as Azure CLI and Azure PowerShell to handle the new API responses.
+1. Collaborate with the Portal team to update the deployments blade, ensuring that the UI reflects the API changes.
+
+## Unresolved questions
+
+### Key vault secret expressions in extension configurations and other expression limitations driven by stacks
+
+Whether key vault secret access should be supported in extension configurations is a tricky subject. Users will likely
+want this feature, and it would work fine with a standalone deployment that isn't tethered to a stack because the secret
+is only needed at the time of deployment. The problem from the stacks perspective with introducing this feature is 
+that accessing a user's key vault in the future for the deletion of stack resources introduces a failure point. What if
+the user did not properly maintain this secret and it's now invalid? This type of issue would block the deletion or 
+update of a stack when resources that no longer managed are to be deleted and potentially result in unexpected stack
+states. The errors would likely manifest as incidents for the stacks team to investigate.
```

Comment:
This feels like a problem that can't be solved by stacks. If the data is gone, it's gone. For example in your AKS scenario - I could go out of band and delete the cluster 👿 

What's the right behavior when this happens?